src/ILLink.Shared/DiagnosticId.cs

```diff
@@ -187,7 +187,11 @@ public enum DiagnosticId
 		// Dynamic code diagnostic ids.
 		RequiresDynamicCode = 3050,
 		RequiresDynamicCodeAttributeMismatch = 3051,
-		RequiresDynamicCodeOnStaticConstructor = 3052
+		RequiresDynamicCodeOnStaticConstructor = 3052,
+		COMInteropNotSupportedInFullAOT = 3052,
```

Comment:
Notice that there is a collision on the code IL3052, I don't have a strong preference on which one should remain IL3052 and which one should change, but one should be changed.

Reply:
Did we ship one of them in .NET 6? If only one, then we should keep that one as is and change the new one.
If not... I guess just pick one...

Reply:
COMInteropNotSupportedInFullAOT in NativeAOT seems to exist even before runtimelab project existed, so probably I will leave that as 3052 even if we haven't shipped NativeAOT yet